.github/workflows/golint.yaml

```diff
@@ -0,0 +1,29 @@
+name: Go-Linting
+
+on:
+    pull_request: 
+        branches:
+            - main
+        
+    push: 
+        branches:
+            - main
+
+jobs:
+    lint:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Set up Go
+              uses: actions/setup-go@v3
+              with:
+                    go-version: '1.21'  
+
+            - name: Install golangci-lint
```

Comment:
We want all packages to be fetched and setup with bsf and Nix. 
Please check how we've used Taskfile and GitHub Actions to perform a build/test. 
So , 
* Add golangci-lint to dev packages (HCL block). 
* Update taskfile to run it
* Update GitHub Actions to invoke the taskfile .

Reply:
PS: we do need a dedicated workflow file, we can re-use the existing workflow file